### DISCUSSION TITLE
Any way to know the workflow Id before calling GetWorkflowDefinitionAsync

### DISCUSSION THREAD

**Discussion by Alexplose** [elsa-workflows/elsa-core#603] [Q&A]
I designed all my workflow in the designer, so I can't call GetWorkflowDefinition with the workflow class.
I call it using the Id I get in the store.
Is there any way to get this Id by Workflow name or something more friendly ?

**Comment by sfmskywalker** [comment]
Although workflow definitions have a `Name` property, actually intended to be used to query workflow definitions by their name, unfortunately the `IWorkflowDefinitionStore` abstraction does not expose any methods to allow for this.

It should be straightforward to add this method, but of course it will also have to be implemented by all interface implementations.

In Elsa 2, we're getting rid of the repository abstraction in favor of directly exposing YesSQL's `ISession` so that applications can perform any queries they want, while still using a vast array of data stores.

**Comment by sfmskywalker** [comment]
Elsa 2 now has an `IWorkflowRegistry` that allows you to query workflow blueprints (aka definitions), including by name.

E.g.

```csharp
var registry = _serviceProvider.GetRequiredService<IWorkflowRegistry>();
var workflows = await registry.GetWorkflowsAsync();
var myWorkflow = workflows.FirstOrDefault(x => x.Name == "MyWorkflow");
```

I might add a method to allow for passing in a predicate directly, e.g.

```csharp
var myWorkflow = await registry.FindAsync(x => x.Name == "MyWorkflow");
```

That would make it more convenient.
